### ISSUE TITLE
2.5.0 Regression issue with expect().toHaveBeenCalledWith()

### ISSUE THREAD

**Issue by DerFlatulator** [jasmine/jasmine#1188]
After installing 2.5.0, many of our tests are failing with:
```
PhantomJS 2.1.1 (Windows 8 0.0.0) Some test... FAILED
        TypeError: undefined is not a constructor (evaluating 'allKeys[i].match(/^[0-9]+$/)') in C:/<path>/node_modules/jasmine-core/lib/jasmine-core/jasmine.js (line 2988)
        c:/<path>/specFile.js:79:85
```

All the tests I've looked at so far call `toHaveBeenCalledWith`, e.g.:
```js
expect(aSpy).toHaveBeenCalledWith({
  someProperties: "string",
  someOtherProperties: true
})
```

The full function that is throwing an error is:

```js
    function keys(obj, isArray) {
      var allKeys = Object.keys ? Object.keys(obj) :
        (function(o) {
            var keys = [];
            for (var key in o) {
                if (has(o, key)) {
                    keys.push(key);
                }
            }
            return keys;
        })(obj);

      if (!isArray) {
        return allKeys;
      }

      var extraKeys = [];
      for (var i in allKeys) {
        if (!allKeys[i].match(/^[0-9]+$/)) {
          extraKeys.push(allKeys[i]);
        }
      }

      return extraKeys;
    }
```

(edit)

## Temporary Workaround

Lock `jasmine-core` to version `2.4.1` until this is addressed.

**Comment by VolleMelk** [comment]
Same here. The folllowing test fails:

```javascript
it('should equal', function() {
  expect(true).toEqual(true); // ok
  expect([true]).toEqual([true]); // fail
}
```
Fails with
```
PhantomJS 1.9.8 (Windows 8 0.0.0) This should equal FAILED
        TypeError: 'undefined' is not a function (evaluating 'allKeys[i].match(/^[0-9]+$/)')
            at D:/toEqualSpec.js:7
```

**Comment by SuneRadich** [comment]
For now I have downgraded to 2.4.1 and all my tests run again

**Comment by AlexChesters** [comment]
@SuneRadich we've done the same and that solves our failures as well.

**Comment by gebeuhler** [comment]
Am experiencing this issue as well. Will downgrade to 2.4.1.

**Comment by dmcquillan314** [comment]
+1

**Comment by xenobytezero** [comment]
I am also seeing this issue, downgrade has solved it.

This issue only occurs on PhantomJS, In Chrome 53.0.2785.80 I see no issue.

**Comment by chanjunc** [comment]
+1

**Comment by liongoncalves** [comment]
+1

**Comment by chrisfunston** [comment]
+1

**Comment by chrisfunston** [comment]
(Specifically, it's this change: https://github.com/jasmine/jasmine/pull/1155/files)

**Comment by fluffy88** [comment]
Having the same issue and can confirm it's only a problem in PhantomJS, Chrome and Firefox work fine.

**Comment by fluffy88** [comment]
After a little more investigation I found the following.

> INFO: 'AllKeys:', ['0', '1']
INFO: 'I:', '0'
INFO: 'AllKeys[i]:', '0'
INFO: 'I:', '1'
INFO: 'AllKeys[i]:', '1'
INFO: 'I:', 'find'
INFO: 'AllKeys[i]:', function (predicate) { ... }


This issue seems to extend from another PhantomJS issue, where Array.find isn't available in PhantomJS so I am using phantomjs-polyfill-find to add Array.find.

**Comment by logankd** [comment]
I'm seeing this with tests running in Karma and running through the browser as well.
error: " TypeError: allKeys[i].match is not a function"
I see it with toHaveBeenCalledWith as well.

**Comment by pnahtanoj** [comment]
+1

**Comment by jeremyhill-up** [comment]
+1, same deal here (comparing arrays for equality).

**Comment by slackersoft** [comment]
The fix should be released in v2.5.1
